crates/proof-of-sql/src/base/polynomial/interpolate.rs

```diff
@@ -13,7 +13,14 @@ use num_traits::{Inv, One, Zero};
 ///
 /// For any polynomial, `f(x)`, with degree less than or equal to `d`, we have that:
 /// `f(x) = sum_{i=0}^{d} (-1)^(d-i) * (f(i) / (i! * (d-i)! * (x-i))) * prod_{i=0}^{d} (x-i)`
-/// unless x is one of 0,1,...,d, in which case, f(x) is already known.
+/// unless x is one of 0,1,...,d, in which case, f(x) is already known.\
+/// 
+///  # Panics
+/// * This function panics if any of the following occurs:
+///   * When computing the inverse of `(i! * (d-i)! * (x-i))`, if the value `x - i` is zero
+///     (which is guaranteed to not happen as `i == x` will return early).
```

Comment:
Since this is guaranteed to not happen, it shouldn't be included in the documentation.

Instead, the `unwrap` should really be an `expect` with a reason.